DirectEmployers/MyJobs
# Some multifilter analytics API

## What's this PR do?

Modifies the `dynamic` api in Analytics and the front end to provide better sense of where we are in the dimension list when querying. It also adds a `multifilter` when one is supposed to be displayed.

## Where should the reviewer start?

The new report schema changes I made are worth questioning. These are the bulk of the changes.

`analytics/api.py` contains most of the back end code for the new multifilter control. That control also requires the "group_by" column to be a plural entity.

Most of the changes are a result of the two items above.

## How should this be manually tested?

Set the "new-site-traffic" switch.

`make_data` generates site traffic test data so you can use that to see working charts.

Make sure that the analytics for traffic in particular aren't broken.

Check that exports still work.

Check the "Source" and "Technology" 

Turn off the "new-site-analytics" switch and make sure that the old site traffic analytics dimensions still work.

## Any background context you want to provide?
